api/REST-API.md

```diff
@@ -1,90 +1,96 @@
+If the request passes through a cache and the Request-URI identifies one or more currently cached entities, those entries SHOULD be treated as stale. Responses to this method are not cacheable.
+
+ Objects in REST are always manipulated from the URI. It is the URI and no other element that is the sole identifier of each resource in this REST system. The URI allows us to access the information in order to change or delete it, or for example to share its exact location with third parties.  
+ 
+ **Advantages of REST for developement:**
+  - Separation between the client and the server: the REST protocol totally separates the user interface from the server and the data storage. This has some advantages when making developments. For example, it improves the portability of the interface to other types of platforms, it increases the scalability of the projects, and allows the different components of the developments to be evolved independently.
+  - Visibility, reliability and scalability. The separation between client and server has one evident advantage, and that is that each development team can scale the product without too much problem. They can migrate to other servers or make all kinds of changes in the database, provided the data from each request is sent correctly. The separation makes it easier to have the front and the back on different servers, and this makes the apps more flexible to work with.
+  -  The REST API is always independent of the type of platform or languages: the REST API always adapts to the type of syntax or platforms being used, which gives considerable freedom when changing or testing new environments within the development. With a REST API you can have PHP, Java, Python or Node.js servers. The only thing is that it is indispensable that the responses to the requests should always take place in the language used for the information exchange, normally XML or JSON. 
 
 ### Guided Practice
 
-Have the apprentices work with you as you do something step-by-step.  This can also be fulfilled by a detailed tutorial intended for beginners.
-
+Complete the Links mentioned in the Materials sections. Follow step by step whatever mentioned in those turorials and try to cover it all.
 
 ### Independent Practice
 
-Class does this thing themselves with specific additional items. This could be alone, with a partner, or small group; but the idea is that it's less guided, more independent.
-
-### Challenge
+Go through the link and follow the tutorial:
+ - [Learn Nodejs](https://www.w3schools.com/nodejs/)
```

Comment:
If you cannot find a tutorial similar to you first one, but using sql and postgresql instead of mongo, the alternative is to keep that tutorial and give very clear directions about how to do it with sql instead.

Reply:
I hope you would @alodahl I have added a new tutorial similar to the first one. If this won't work then I shall Place the first one.

Reply:
@alodahl  I have added a new tutorial , have a look on it. If that won't work I will place the first one.